indexing-service/src/main/java/io/druid/indexing/common/actions/SegmentNukeAction.java

```diff
@@ -62,7 +62,7 @@ public SegmentNukeAction(
   @Override
   public Void perform(Task task, TaskActionToolbox toolbox) throws IOException
   {
-    toolbox.verifyTaskLocks(task, segments);
+    TaskActionPreconditions.checkTaskLocks(task, toolbox.getTaskLockbox(), segments);
```

Comment:
Good catch! I added lock upgrade here and ```SegmentMetadataUpdateAction```. Additionally, some other actions like ```SegmentListUsedAction``` and ```SegmentListUnusedAction``` also look to check lock before performing their actions. I'll check and raise a new pr for this issue.